packages/grid/x-grid-modules/src/utils/sortingUtils.ts

```diff
@@ -22,7 +22,7 @@ export const nillComparer = (v1: CellValue, v2: CellValue): number | null => {
 
 export const stringNumberComparer: ComparatorFn = (v1: CellValue, v2: CellValue) => {
   const nillResult = nillComparer(v1, v2);
-  if (nillResult != null) {
+  if (nillResult !== null) {
     return nillResult;
   }
```

Comment:
The method seems to be relatively slow when sorting 100,000 rows in memory. I'm not sure we have any great alternatives.